src/libertem/udf/stddev.py

```diff
@@ -0,0 +1,189 @@
+    the corresponding BufferWrapper objects
+    """
+    return {
+        'var': BufferWrapper(
+            kind='sig', dtype='float32'
+            ),
+        'std': BufferWrapper(
+            kind='sig', dtype='float32'
+            ),
+        'mean': BufferWrapper(
+            kind='sig', dtype='float32'
+            ),
+        'num_frame': BufferWrapper(
+            kind='single', dtype='float32'
+            ),
+        'sum_frame': BufferWrapper(
+            kind='sig', dtype='float32'
+            )
+    }
+
+
+def compute_batch(frame, var, std, mean, sum_frame, num_frame):
+    """
+    Given a frame, update sum of variances, sum of frames,
+    and the number of total frames
+
+    Parameters
+    ----------
+    frame
+        single frame of the data
+
+    var
+        Buffer that stores sum of variances of the previous set of frames
+
+    sum_frame
+        Buffer that sores sum of frames of the previous set of frames
+
+    num_frame
+        Buffer that stores the number of frames used for computation
+
+    """
+    if num_frame == 0:
+        var[:] = 0
+
+    else:
+        p0 = VariancePart(var=var, sum_im=sum_frame, N=num_frame)
+        p1 = VariancePart(var=0, sum_im=frame, N=1)
+        compute_merge = merge(p0, p1)
+
+        var[:] = compute_merge.var
+
+    sum_frame[:] += frame
+    num_frame[:] += 1
+
+
+def batch_merge(dest, src):
+    """
+    Given two buffers that contain sum of variances, sum of frames,
+    and the number of frames used in each of the partitions, merge the
+    partitions and compute the joint sum of variances and sum of frames
+    over all frames used
+
+    Parameters
+    ----------
+    dest
+        A buffer that contains sum of variances, sum of frames, and the
+        number of frames used over all the frames used
+
+    src
+        A buffer that contains sum of variances, sum of frames, and the
+        number of frames used over current iteration of partition
+    """
+    p0 = VariancePart(var=dest['var'][:],
+                    sum_im=dest['sum_frame'][:],
+                    N=dest['num_frame'][:])
+    p1 = VariancePart(var=src['var'][:],
+                    sum_im=src['sum_frame'][:],
+                    N=src['num_frame'][:])
+    compute_merge = merge(p0, p1)
+
+    dest['var'][:] = compute_merge.var
+    dest['sum_frame'][:] = compute_merge.sum_im
+    dest['num_frame'][:] = compute_merge.N
+
+
+def merge(p0, p1):
+    """
+    Given two sets of partitions, with sum of frames
+    and sum of variances, compute joint sum of frames
+    and sum of variances using one pass algorithm
+
+    Parameters
+    ----------
+    p0
+        Contains information about the first partition, including
+        sum of variances, sum of pixels, and number of frames used
+
+    p1
+        Contains information about the second partition, including
+        sum of variances, sum of pixels, and number of frames used
+
+    Returns
+    -------
+    collections.namedtuple object
```

Comment:
`collections.namedtuple` is rather generic, the specific returned object is a `VariancePart`

Reply:
fixed it!